### PR TITLE
DI-311 Report logging in event processor for invalid open times

### PR DESCRIPTION
**Pull request by SLClayton**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-311

## Description

Added report logging to splunk for invalid opening times.

### Noteworthy Changes

- Initial check of invalid opening times for NHS Entity object for case of logging (subsequent checks used to skip adding to change req)
- Unit tests to run on scenario of invalid opening times entered.

## Type of change

Delete not appropriate

- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [x] Unit tests
- [ ] Integration tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by mabe13 — APPROVED** [review]
LGTM 👍 
Just edited the PR title to include the ticket number

**Review by rohith1122 — APPROVED** [review]
LGTM. Wait for the testing team approval as well.

**Pull request merged**
